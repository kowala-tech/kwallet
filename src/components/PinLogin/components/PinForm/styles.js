const styles = (theme) => ({
	root: {
	},
	input: {
		position: "absolute",
		top: -1000
	},
	pinDigit: {
		marginRight: theme.spacing.unit * 2,
		marginLeft: theme.spacing.unit * 2
	}
});

export default styles;
