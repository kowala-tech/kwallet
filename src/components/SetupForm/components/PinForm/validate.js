const validate = values => {
	const errors = {};
	// Pin
	if (!values.pin) {
		errors.pin = "PIN is required.";
	} else if (values.pin.length !== 4) {
		errors.pin = "PIN must 4 digits.";
	}
	// Pin Confirmation
	if (values.pin !== values.pinConfirmation) {
		errors.pinConfirmation = "PINs do not match.";
	}
	return errors;
};

export default validate;
