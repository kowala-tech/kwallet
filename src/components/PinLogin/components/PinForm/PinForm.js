import React from "react";
import PropTypes from "prop-types";
import { compose } from "recompose";
import classNames from "classnames";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import RadioButtonChecked from "@material-ui/icons/RadioButtonChecked";
import RadioButtonUnchecked from "@material-ui/icons/RadioButtonUnchecked";
// Component Related Imports
import styles from "./styles";

class PinForm extends React.Component {

	constructor(props) {
		super(props);
		this.pinInput = React.createRef();
		this.state = { pin: "" };
	}

	componentDidMount() {
		this.focus();
	}

	focus() {
		this.pinInput.current.focus();
	}

	handleChange(event) {
		this.setState(
			{ pin: event.target.value },
			() => {
				if (this.state.pin.length === 4) {
					this.submit();
				}
			}
		);
	}

	clearPin() {
		this.setState(
			{ pin: "" },
			() => {
				this.pinInput.current.value = "";
			}
		);
	}

	submit() {
		this.props.onSubmit( this.props.username, this.state.pin, this.clearPin.bind(this) );

	}

	render() {
		const {
			classes,
			loading,
			error
		} = this.props;

		let pinDigit = (pinLength, delay) => {
			let className = classNames(
				"kowala",
				[classes.pinDigit],
				{
					animated: loading,
					bounce: loading,
					infinite: loading
				}
			);
			return(
				loading || this.state.pin.length >= pinLength ?
					<RadioButtonChecked
						className={className}
						style={{ animationDelay: delay }}
					/>
					: <RadioButtonUnchecked
						className={className}
					/>
			);
		};

		return (
			<div
				className={classes.root}
				onClick={this.focus.bind(this)}
			>
				<div className={classNames({ "animated": error, "headShake": error })}>
					{pinDigit(1,"0.0s")}
					{pinDigit(2,"0.2s")}
					{pinDigit(3,"0.4s")}
					{pinDigit(4,"0.6s")}
				</div>
				<form>
					<input
						ref={this.pinInput}
						type="number"
						className={classes.input}
						onChange={this.handleChange.bind(this)}
					/>
				</form>
			</div>
		);
	}
}

PinForm.propTypes = {
	classes: PropTypes.object.isRequired,
	onSubmit: PropTypes.func.isRequired,
	loading: PropTypes.bool.isRequired,
	error: PropTypes.bool.isRequired,
	username: PropTypes.string.isRequired
};

export default compose(
	withStyles(styles)
)(PinForm);
