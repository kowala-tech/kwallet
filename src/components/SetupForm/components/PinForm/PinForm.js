import React from "react";
import { compose } from "recompose";
import PropTypes from "prop-types";
// Material UI Imports
import { withStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
// Form Imports
import { Field, reduxForm } from "redux-form";
import validate from "./validate";
import ReduxFormField from "../../../ReduxFormField";
// Component Related Imports
import styles from "../../styles";

class PinForm extends React.Component {
	render() {
		const { classes, handleSubmit, invalid, pristine, previousStep } = this.props;
		return (
			<form onSubmit={handleSubmit}>
				<Field
					name="pin"
					type="password"
					component={ReduxFormField}
					label="PIN"
				/>
				<Field
					name="pinConfirmation"
					type="password"
					component={ReduxFormField}
					label="PIN Confirmation"
				/>
				<div className={classes.actionsContainer}>
					<Button
						onClick={previousStep}
						className={classes.button}
					>
						Back
					</Button>
					<Button
						variant="raised"
						color="primary"
						className={classes.button}
						type="submit"
						disabled={invalid || pristine}
					>
						Finish
					</Button>
				</div>
			</form>
		);
	}
}

PinForm.propTypes = {
	classes: PropTypes.object.isRequired,
	onSubmit: PropTypes.func.isRequired,
	previousStep: PropTypes.func.isRequired,
	handleSubmit: PropTypes.func.isRequired,
	invalid: PropTypes.bool.isRequired,
	pristine: PropTypes.bool.isRequired,
};

export default compose(
	withStyles(styles),
	reduxForm({
		form: "setupForm",
		destroyOnUnmount: false,
		forceUnregisterOnUnmount: true,
		validate
	})
)(PinForm);
