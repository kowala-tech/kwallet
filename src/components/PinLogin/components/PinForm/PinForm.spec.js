import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
// Story related component imports
import PinForm from "./PinForm";

describe("PinForm", () => {

	it("it should match the snapshot", () => {
		const component = toJson(
			shallow(
				<PinForm
					onSubmit={jest.fn()}
					loading={false}
					error={false}
					username={"test-user"}
				/>
			)
		);
		expect(component).toMatchSnapshot();
	});

});
