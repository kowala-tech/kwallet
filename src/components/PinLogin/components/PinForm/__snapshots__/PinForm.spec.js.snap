// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinForm it should match the snapshot 1`] = `
<PinForm
  classes={
    Object {
      "input": "PinForm-input-2",
      "pinDigit": "PinForm-pinDigit-3",
      "root": "PinForm-root-1",
    }
  }
  error={false}
  loading={false}
  onSubmit={[MockFunction]}
  username="test-user"
/>
`;
